src/Symfony/Component/DependencyInjection/Compiler/ValidateEnvPlaceholdersInExtensionConfigPass.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Compiler;
+
+use Symfony\Component\Config\Definition\BaseNode;
+use Symfony\Component\Config\Definition\Processor;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Extension\ConfigurationExtensionInterface;
+use Symfony\Component\DependencyInjection\ParameterBag\EnvPlaceholderParameterBag;
+use Symfony\Component\DependencyInjection\ParameterBag\ParameterBag;
+
+/**
+ * Validates environment variable placeholders used in extension configuration with dummy values.
+ *
+ * @author Roland Franssen <franssen.roland@gmail.com>
+ */
+class ValidateEnvPlaceholdersInExtensionConfigPass implements CompilerPassInterface
+{
+    private static $typeFixtures = array('array' => array(), 'bool' => false, 'float' => 0.0, 'int' => 0, 'string' => '');
+
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        if (!class_exists(BaseNode::class) || !($extensions = $container->getExtensions())) {
+            return;
+        }
+
+        $resolvingBag = $container->getParameterBag();
+        if (!$resolvingBag instanceof EnvPlaceholderParameterBag) {
+            return;
+        }
+
+        $defaultBag = new ParameterBag($container->getParameterBag()->all());
+        $envTypes = $resolvingBag->getProvidedTypes();
+        foreach ($resolvingBag->getEnvPlaceholders() + $resolvingBag->getUnusedEnvPlaceholders() as $env => $placeholders) {
+            $prefix = (false === $i = strpos($env, ':')) ? 'string' : substr($env, 0, $i);
+            if (!($types = isset($envTypes[$prefix]) ? $envTypes[$prefix] : array('string'))) {
+                continue;
+            }
+            $default = ($hasEnv = (false === $i && $defaultBag->has("env($env)"))) ? $defaultBag->get("env($env)") : null;
+//            if (null !== $default && !in_array($type = gettype($default), $types, true)) {
+//                throw new LogicException(sprintf('Invalid type for env parameter "env(%s)", expected "%s" got "%s".', $env, implode('", "', $types), $type));
+//            }
+            $values = array();
+            foreach ($types as $type) {
+                $values[$type] = $hasEnv ? $default : (isset(self::$typeFixtures[$type]) ? self::$typeFixtures[$type] : null);
+            }
+            foreach ($placeholders as $placeholder) {
+                BaseNode::setPlaceholder($placeholder, $values);
+            }
+        }
+
+        // @todo
+        //BaseNode::placeholdersOnly();
```

Comment:
basically im not sure if we should care / expect a heavy config tier or so (closures invoking twice). As currently we do a full second run of validation, where it could be partial (placeholders only). Not sure if worth it :)

Reply:
let's not care :)